### PR TITLE
[RLlib] Bandits use TrainerConfig objects.

### PR DESCRIPTION
**Pull request by smorad**
## Why are these changes needed?

This uses the new TrainerConfig paradigm with contextual bandits
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/bandit/bandit.py [hunk 1]
```diff
@@ -1,49 +1,102 @@
...
 
-    # Make sure, a `train()` call performs at least 100 env sampling timesteps, before
-    # reporting results. Not setting this (default is 0) would significantly slow down
-    # the Bandit Trainer.
-    "min_sample_timesteps_per_reporting": 100,
-})
-# __sphinx_doc_end__
-# fmt: on
+    Example:
+        >>> config = BanditConfig().rollouts(num_rollout_workers=4).
```
Comment (maxpumperla) [inline rllib/agents/bandit/bandit.py:24]: hey @smorad thanks for this PR! Can we please get the resp. `BanditConfig` import here?

File: rllib/agents/bandit/bandit.py [hunk 2]
```diff
@@ -1,49 +1,102 @@
...
+        self.min_sample_timesteps_per_reporting = 100
+        # __sphinx_doc_end__
+        # fmt: on
+
+
+class BanditLinTSConfig(BanditConfig):
+    """Defines a contextual bandit configuration class from which
+    a Thompson-sampling contexual bandit can be built.
+    Example:
+        >>> config = BanditLinTSConfig().rollouts(num_rollout_workers=4)
```
Comment (maxpumperla) [inline rllib/agents/bandit/bandit.py:55]: here, too

File: rllib/agents/bandit/bandit.py [hunk 3]
```diff
@@ -1,49 +1,102 @@
...
+        self.exploration_config = {"type": "ThompsonSampling"}
+        # __sphinx_doc_end__
+        # fmt: on
+
+
+class BanditLinUCBConfig(BanditConfig):
+    """Defines a contextual bandit configuration class from which
+    a upper confidence bound contexual bandit can be built.
+    Example:
+        >>> config = BanditLinUCBConfig().rollouts(num_rollout_workers=4)
```
Comment (maxpumperla) [inline rllib/agents/bandit/bandit.py:76]: and here as well

File: rllib/agents/bandit/bandit.py (hunk 1 above)
Comment (smorad) [reply rllib/agents/bandit/bandit.py]: It isn't python to have two separate ways to do the same thing. Example removed and users should now use subclasses instead (which each have correct imports now).

**Review by sven1977 — APPROVED** [review]
LGTM. Thanks for this PR @smorad !

On the removed example: True, shouldn't do one thing in two different ways in python, but RLlib has been supporting running with Tune or "manually" with Trainer for ages :) And Tune doesn't support the new config objects (yet(!) :) ).

**Pull request merged**